src/ore/src/id_gen.rs

```diff
@@ -110,27 +111,25 @@ where
 #[derive(Clone, Debug)]
 pub enum IdHandle<T: From<u8> + AddAssign + PartialOrd + Copy> {
     /// An ID "allocated" a compile time.
+    ///
+    /// Note: It is *entirely* up to the caller to make sure the provided ID is
+    /// not used by a dynamic ID allocator.
     Static(T),
-    /// An ID allocated at runtime, get's freed once all handles have been dropped.
+    /// An ID allocated at runtime, gets freed once all handles have been dropped.
     Dynamic(Arc<internal::IdHandleInner<T>>),
 }
 
 impl<T> IdHandle<T>
 where
     T: From<u8> + AddAssign + PartialOrd + Copy,
 {
-    /// Creates a new [`IdHandle`] with the provided ID.
+    /// Returns the raw ID inside of this handle.
     ///
-    /// Note: It is *entirely* up to the caller to make sure the provided ID is
-    /// not already being used.
-    pub const fn new_static(id: T) -> Self {
```

Comment:
This wasn't technically related to this change, but I noticed it while I was in here and it seemed confusing to have a constructor method that was equivalent to the native enum constructor.